src/cache.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+// TODO: move to its own package
+
+import fs from 'fs-extra'
+import moment from 'moment'
+
+export default class {
+  static async fetch (cachePath: string, cacheDuration: number, options: any = {}): Promise<?Array<?string>> {
+    let cache
+    let cachePresent = await fs.exists(cachePath)
+    if (cachePresent) {
+      if (this._isStale(cachePath, cacheDuration)) {
+        // TODO: move this to a fork
+        let cache = await options.cacheFn()
+        this._updateCache(cachePath, cache)
```

Comment:
@RasPhilCo could you await this? I know it will slow it down, but it feels better than possibly clipping it and ending up with invalid JSON